### DISCUSSION TITLE
Allocate outside of run function

### DISCUSSION THREAD

**Discussion by steckes** [pykeio/ort#37] [Ideas]
Hey!
For the usage in audio processes and in general for the performance it would be really nice if the inputs and outputs could be set and allocated before the actual run() function. Somewhat like suggested here: 
https://github.com/nbigaouette/onnxruntime-rs/pull/41

This could be an additional function as well. Might this be something you would be interested in as well?

And thank you, I am very happy that someone is still working on a onnxruntime wrapper in rust, the crate is working out of the box like a charm for me :)

**Comment by decahedron1** [comment]
Tracking this request (and others) in #38. I hope to have this implemented in ort 1.15, which may release near the end of the quarter depending on when Microsoft releases ONNX Runtime 1.15.

**Comment by steckes** [comment]
Onnx Runtime 1.15 is out :)
I saw that you made a lot of progress already on the I/O rework, amazing!
Do you already know when you will have time to finish the 1.15 update?
